src/ast/ast_rewrite_star_projections.c

```diff
@@ -0,0 +1,332 @@
+/*
+* Copyright 2018-2022 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "ast.h"
+#include "../query_ctx.h"
+#include "../errors.h"
+#include "../util/arr.h"
+#include "../util/qsort.h"
+#include "../util/sds/sds.h"
+#include "../procedures/procedure.h"
+
+//------------------------------------------------------------------------------
+//  Annotation context - WITH/RETURN * projections
+//------------------------------------------------------------------------------
+static void _collect_aliases_in_path
+(
+	const cypher_astnode_t *path,
+	const cypher_astnode_t ***identifiers
+) {
+	uint path_len = cypher_ast_pattern_path_nelements(path);
+	// Every even offset corresponds to a node.
+	for(uint i = 0; i < path_len; i += 2) {
+		const cypher_astnode_t *ast_node = cypher_ast_pattern_path_get_element(path, i);
+		const cypher_astnode_t *ast_alias = cypher_ast_node_pattern_get_identifier(ast_node);
+		if(ast_alias == NULL) continue;  // Unaliased node, do nothing.
+		// Add node alias to projection array.
+		array_append(*identifiers, ast_alias);
```

Comment:
It does, but I'm dereferencing here in case `array_grow` triggers a realloc (in which case only the passed-by-value pointer in this stack frame will be updated).